nasa/harmony-swath-projector
# TRT-558 - Migrate to earthdata-varinfo==3.0.0.

## Description

This PR updates the Swath Projector to use `earthdata-varinfo==3.0.0`. This version of the `earthdata-varinfo` package includes significant updates to the configuration file schema and some of the publicly accessible methods and attributes on classes contained in the package.

In this service, it looks like that none of the updated class attributes and methods were being called, so the only updates were to the configuration file itself (mainly removing a couple of underscores and renaming `CF_Overrides` to `MetadataOverrides`). I've also renamed that configuration file for clarity.

## Jira Issue ID

TRT-558.

## Local Test Steps

* Pull this branch.
* Check the `CHANGELOG.md` to make sure I got the updated format right.
* Run the tests: `./bin/buid-image && ./bin/build-test && ./bin/run-test`.
* Check that the new version of the configuration file is compliant with the `earthdata-varinfo` schema (will require the `jsonschema` Python package):

```
import json

from jsonschema import validate

with open('/path/to/swath_projector/earthdata_varinfo_config.json') as file_handler:
    sp_config = json.load(file_handler)

with open('/path/to/earthdata-varinfo/config/1.0.0/earthdata_varinfo_configuration_schema.json') as file_handler:
    ev_schema = json.load(file_handler)

validate(schema=ev_schema, instance=sp_config)

# If the configuration file adheres to the schema, there will be no output in the Python repl. If the configuration file does not, then errors will be printed out to the terminal.
```

## PR Acceptance Checklist
* [x] Jira ticket acceptance criteria met. (Specifically the Swath Projector bit)
* [x] Tests added/updated and passing. (Not updated, just passing)
